extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1388,6 +1388,47 @@ describes.realWin('amp-story', {
                 expect(story.activePage_.element.id).to.equal('cover');
               });
         });
+
+    it('should correctly mark goToPage pages are distance 1', () => {
+      createPages(
+          story.element, 4, ['cover', 'page-1', 'page-2', 'page-3']);
+      story.buildCallback();
+
+      return story.layoutCallback()
+          .then(() => {
+            story.element.setAttribute('id', 'story');
+
+            const actionButton = createElementWithAttributes(
+                win.document,
+                'button',
+                {'id': 'actionButton',
+                  'on': 'tap:story.goToPage(id=page-2)'});
+
+            story.element.querySelector('#cover').appendChild(actionButton);
+
+            const distanceGraph = story.getPagesByDistance_();
```

Comment:
Nit: Can we set the goToPage action in the page before the buildCallback, so we don't have to manually call this private method?

Reply:
I had some issues retrieving the distance attribute from the pages for some reason (it doesn't seem to exist during the tests on the page elements during the tests, though it's definitely being set). If it's okay with you, I think we should keep the as they are.